lib/rester/middleware/correlation_id.rb

```diff
@@ -0,0 +1,16 @@
+module Rester
+  module Middleware
+    ##
+    # Sets the correlation id for the current thread if one is passed in the
+    # request header. Clean up the correlation id for this thread before
+    # responding.
+    class CorrelationId < Base
+      def call(env)
+        Rester.correlation_id = env['X-Rester-Correlation-ID']
+        super.tap { |response|
+          Rester.correlation_id = nil
```

Comment:
Actually, this should be done in an `ensure` block, in case `super` raises an error.  We could have a memory leak in that the `correlation_id` hash could fill up without being cleaned.

```ruby
def call(env)
  Rester.correlation_id = env['HTTP_RESTER_CORRELATION_ID']
  super
ensure
  Rester.correlation_id = nil
end
```